### ISSUE TITLE
Unable to process $XRD_PLUGINCONFDIR if user has no password entry

### ISSUE THREAD

**Issue by bbockelm** [xrootd/xrootd#1229]
If a user has no `/etc/passwd` entry then the environment variable `$XRD_PLUGINCONFDIR` is not processed:

https://github.com/xrootd/xrootd/blob/master/src/XrdCl/XrdClPlugInManager.cc#L229-L234

In our case, this occurs because we are using a container and have no clue what UID the container would be launched as -- and use a non-standard location for the plugin directory because we are installing via conda.

**Comment by abh3** [comment]
@bbockelm Well, I am as mystified as you as what the correct solution would be. Can you provide one or more suggestions?

**Comment by bbockelm** [comment]
Hi Andy,

I think something like this would suffice:

```
      passwd *pwd = pwdHandler.Get( getuid() );
      if( pwd ) {
         std::string userPlugIns = pwd->pw_dir;
         userPlugIns += "/.xrootd/client.plugins.d";
         ProcessConfigDir( userPlugIns );
      }
```

That is, continue processing instead of erroring out.

**Comment by simonmichal** [comment]
Fixed in 597509497b59608bf8c0f104bdec5d1b2a5c29b2
